From 8c541698365ee5d7c36f304be2ee8c60ae028190 Mon Sep 17 00:00:00 2001
From: "R. Voggenauer" <rvogg@users.noreply.github.com>
Date: Tue, 14 Feb 2023 15:21:06 +0100
Subject: [PATCH] Stop using access time for file comparison.

The comparison of access times was introduced with commit 1e39ef331b2b78baec84fdb577d497511cc46bd5.
This leads to the fact that the lld sets the variable isUnderSysroot to false by mistake with strongly
parallelized compiles. (Since the access time of the sysroot folder can be updated between two stats).

This prevents the ScriptParser from correctly resolving the path from a linker script if sysroot is set.
---
 llvm/lib/Support/Windows/Path.inc | 2 --
 1 file changed, 2 deletions(-)

diff --git a/llvm/lib/Support/Windows/Path.inc b/llvm/lib/Support/Windows/Path.inc
index 45f0e95c87fe..b949b724509f 100644
--- a/llvm/lib/Support/Windows/Path.inc
+++ b/llvm/lib/Support/Windows/Path.inc
@@ -650,8 +650,6 @@ bool equivalent(file_status A, file_status B) {
   return A.FileIndexHigh == B.FileIndexHigh &&
          A.FileIndexLow == B.FileIndexLow && A.FileSizeHigh == B.FileSizeHigh &&
          A.FileSizeLow == B.FileSizeLow &&
-         A.LastAccessedTimeHigh == B.LastAccessedTimeHigh &&
-         A.LastAccessedTimeLow == B.LastAccessedTimeLow &&
          A.LastWriteTimeHigh == B.LastWriteTimeHigh &&
          A.LastWriteTimeLow == B.LastWriteTimeLow &&
          A.VolumeSerialNumber == B.VolumeSerialNumber;
-- 
2.39.1

